packages/ra-ui-materialui/src/input/TextInput.tsx

```diff
@@ -49,7 +49,7 @@ export const TextInput = (props: TextInputProps) => {
     } = useInput({
         defaultValue,
         format,
-        parse,
+        parse: parse ?? defaultParse,
```

Comment:
I believe that a textInput should set the value to `null` when emptied. 

I believe it should be the same for e.g. `<SelectInput>`(which already does that when inside a `<ReferenceInput>`, and several other inputs, too. What do you think? I which case would we want an empty form value *not*  to translate to `null`?

Reply:
I agree. Hence I generalized the solution to all inputs.